cpp/src/parquet/arrow/schema.cc

```diff
@@ -427,6 +428,14 @@ Status FieldToNode(const std::string& name, const std::shared_ptr<Field>& field,
     }
     case ArrowTypeId::EXTENSION: {
       auto ext_type = std::static_pointer_cast<::arrow::ExtensionType>(field->type());
+      // Built-in JSON extension is handled differently.
+      if (ext_type->extension_name() ==
+          std::static_pointer_cast<::arrow::ExtensionType>(::arrow::extension::json())
+              ->extension_name()) {
```

Comment:
You could just hard-code `std::string("arrow.json")` here.

Reply:
Done.